jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1052,14 +1065,22 @@ else if (_endOfContent == EndOfContent.CHUNKED_CONTENT)
                     if (_fieldCache == null)
                     {
                         _fieldCache = (getHeaderCacheSize() > 0 && (_version != null && _version == HttpVersion.HTTP_1_1))
-                            ? new ArrayTernaryTrie<>(getHeaderCacheSize())
+                            ? new Index.Builder<HttpField>()
+                            .caseSensitive(false)
+                            .mutable()
+                            .maxCapacity(getHeaderCacheSize())
+                            .build()
                             : NO_CACHE;
                     }
 
-                    if (!_fieldCache.isFull())
+                    if (_field == null)
+                    {
+                        HttpField field = new HttpField(_header, caseInsensitiveHeader(_headerString, _header.asString()), _valueString);
+                        if (_fieldCache.put(field))
+                            _field = field;
+                    }
+                    else
```

Comment:
Hmmmm not having isFull changes this a bit... plus I;m not sure it is right.  Maybe something like:
```suggestion
                    if (_field == null)
                        _field = new HttpField(_header, caseInsensitiveHeader(_headerString, _header.asString()), _valueString);
                    if (!_fieldCache.put(field))
                    {
                        _fieldCache.clear();
                        _fieldCache.put(field);
                    }
```

Reply:
Ideally, we should rely on some eviction algorithm in the index instead of clearing it. But that's quite a can of worm that should be left closed if possible.

Reply:
I think before it just filled up the cache and left it as is.... so maybe we just put and ignore the return for now.... we can refine in future PRs